src/CoreBundle/Resources/public/javascript/modules/sidebar/sidebar.js

```diff
@@ -42,6 +43,21 @@
             $matchingNavItems.removeClass('d-none');
             $matchingNavItems.parents('.nav-item').addClass('open').removeClass('d-none');
         },
+        onCategoryToggle: function (event) {
+            const $category = $(event.target).parent('.nav-dropdown');
+            this.$baseElement.find('.nav-dropdown.open').not($category).removeClass('open');
+
+            let categoryId = null;
+            // The following condition is inverted, as this handler will be executed before the actual class change.
+            if (false === $category.hasClass('open')) {
+                categoryId = $category.data('categoryid');
+            }
+
+            const url = this.$baseElement.data('save-active-category-notification-url');
+            $.post(url, {
+                categoryId: categoryId
```

Comment:
errors are not handled here?

Reply:
We can't really do anything in case an error occurs. The sidebar will still work on failure (only the state won't be saved), so it's better not to bother the user,

Reply:
You could log an error to the console.

Reply:
Logging to console in production code is considered bad practice. I'd just leave it as is: In case of an error the sidebar will still mostly work, and in development the error can easily be seen as failed request in the debug toolbar.